### PR TITLE
Better deadLetter logging of wrapped messages, #28109

### PR DESCRIPTION
**Pull request by patriknw**
* specifically AdaptMessage that is used for Typed messageAdapter
  and ActorContext.ask

Refs #28109

### CODE REVIEW

File: akka-actor/src/main/scala/akka/actor/ActorRef.scala [hunk 1]
```diff
@@ -533,6 +533,28 @@ object Dropped {
...
+    }
+  }
+}
+
+/**
+ * Message envelopes may implement this trait for better logging, such as logging of
+ * message class name of the wrapped message instead of the envelope class name.
+ */
+trait WrappedMessage {
+  def message: Any
```
Comment (hepin1989) [inline akka-actor/src/main/scala/akka/actor/ActorRef.scala]: rawMessage or raw?

File: akka-remote/src/main/scala/akka/remote/RemoteMetricsExtension.scala [hunk 2]
```diff
@@ -70,7 +70,7 @@ private[akka] class RemoteMetricsOn(system: ExtendedActorSystem) extends RemoteM
   override def logPayloadBytes(msg: Any, payloadBytes: Int): Unit =
     if (payloadBytes >= logFrameSizeExceeding) {
       val clazz = msg match {
-        case x: ActorSelectionMessage => x.msg.getClass
+        case x: ActorSelectionMessage => x.message.getClass
```
Comment (hepin1989) [inline akka-remote/src/main/scala/akka/remote/RemoteMetricsExtension.scala]: +1

**Review by chbatey — APPROVED** [review]
Looks like a good idea

File: akka-actor/src/main/scala/akka/actor/ActorRef.scala [hunk 3]
```diff
@@ -533,6 +533,28 @@ object Dropped {
...
+}
```
Comment (hepin1989) [inline akka-actor/src/main/scala/akka/actor/ActorRef.scala]: Would it be better to extract this trait to a dedicated file?

Comment (patriknw) [reply akka-actor/src/main/scala/akka/actor/ActorRef.scala]: we already have similar/related here, such as DeadLetter and Dropped

Comment (hepin1989) [reply akka-actor/src/main/scala/akka/actor/ActorRef.scala]: akka.cluster.pubsub.WrappedMessage

**Review by johanandren — APPROVED** [review]
LGTM

**Pull request merged**